server/etcdserver/api/membership/store.go

```diff
@@ -64,14 +71,29 @@ func TrimClusterFromBackend(be backend.Backend) error {
 	return nil
 }
 
-func mustDeleteMemberFromBackend(be backend.Backend, id types.ID) {
+func unsafeDeleteMemberFromBackend(be backend.Backend, id types.ID) error {
 	mkey := backendMemberKey(id)
 
 	tx := be.BatchTx()
 	tx.Lock()
 	defer tx.Unlock()
-	tx.UnsafeDelete(buckets.Members, mkey)
 	tx.UnsafePut(buckets.MembersRemoved, mkey, []byte("removed"))
+	if !unsafeMemberExists(tx, mkey) {
```

Comment:
Can we document this logic? So, we "dont'" delete the "mkey" from "Members" bucket, and expect "mkey" does not exist?

Reply:
I think that deletion of `mkey` was moved to line 84. Its just not needed if its not found. 

But +1 to providing doc to this and other methods about errors being reported.